### PR TITLE
more psalm; phpcs fixes; fail build on QC issues

### PR DESCRIPTION
**Pull request by pradtke**

### CODE REVIEW

File: lib/Limiters/IpLimiter.php [hunk 1]
```diff
@@ -19,25 +19,32 @@ class IpLimiter extends UserPassBaseLimiter
...
+        if ($this->isIpWhiteListed($this->getClientIp())) {
             return 0;
         }
         return parent::postFailure($username, $password);
     }
 
+    private function getClientIp(): string
+    {
+        /** @var string */
+        return $_SERVER['REMOTE_ADDR'];
```
Comment (tvdijen) [inline lib/Limiters/IpLimiter.php:45]:
```
use Symfony\Component\HttpFoundation\Request;

$request = Request::createFromGlobals();
return $request->server->get('REMOTE_ADDR');
```

This should be more future proof.

File: tests/lib/Limiters/UserPassBaseLimiterTest.php [hunk 2]
```diff
@@ -23,7 +23,7 @@ public function testKeyVariesWithInput(): void
         $this->assertNotEquals($key1, $key2, 'keys should vary');
     }
 }
-
+// phpcs:ignore
```
Comment (tvdijen) [inline tests/lib/Limiters/UserPassBaseLimiterTest.php:26]: Curious:  what does this do? I don't see anything that would trigger an issue

File: lib/Limiters/IpLimiter.php (hunk 1 above)
Comment (pradtke) [reply lib/Limiters/IpLimiter.php:45]: Thanks, I'll make that change in my next update.

File: tests/lib/Limiters/UserPassBaseLimiterTest.php (hunk 2 above)
Comment (pradtke) [reply tests/lib/Limiters/UserPassBaseLimiterTest.php:26]: The test class has a second class (that is used in the test) defined in the file, so this ignore is just saying 'it is okay to have the second class'

**Pull request merged**